code/game/objects/items/weapons/pneumaticCannon.dm

```diff
@@ -29,117 +37,124 @@
 	else
 		if(tank)
 			. += "<span class='notice'>[bicon(tank)] It has \the [tank] mounted onto it.</span>"
-		for(var/obj/item/I in loadedItems)
+		for(var/obj/item/I in loaded_items)
 			. += "<span class='info'>[bicon(I)] It has \the [I] loaded.</span>"
 
+/**
+* Arguments:
+* * I - item to load into the cannon
+* * user - the person loading the item in
+* Returns:
+* * True if item was loaded, false if it failed
+*/
+/obj/item/pneumatic_cannon/proc/load_item(obj/item/I, mob/user)
+	if(I.flags & (ABSTRACT | NODROP | DROPDEL))
+		to_chat(user, "<span class='warning'>You can't put [I] into [src]!</span>")
+		return FALSE
+	if((loaded_weight_class + I.w_class) > max_weight_class)
+		to_chat(user, "<span class='warning'>[I] won't fit into \the [src]!</span>")
+		return FALSE
+	if(I.w_class > src.w_class)
+		to_chat(user, "<span class='warning'>[I] is too large to fit into \the [src]!</span>")
+		return FALSE
+	if(!user.unEquip(I))
```

Comment:
You can probably combine this with the abstract/dropdel check, and/or move that one down here.